xunix/gpu.go

```diff
@@ -176,6 +198,68 @@ func recursiveSymlinks(afs FS, mountpoint string, path string) ([]string, error)
 	return paths, nil
 }
 
+// SameDirSymlinks returns all links in the same directory as `target` that
+// point to target, either indirectly or directly. Only symlinks in the same
+// directory as `target` are considered.
+func SameDirSymlinks(afs FS, target string) ([]string, error) {
+	var (
+		found         = make([]string, 0)
+		maxIterations = 10 // arbitrary upper limit to prevent infinite loops
+	)
+	for range maxIterations {
```

Comment:
This seems like it would be more efficient if it just constructed a map once for every new folder. E.g. `map[string]map[string]string` like `x["/usr/lib"]["nvidia.so"] = "/usr/lib/nvidia.123.so"`, then you could just scan that to determine the symlink chain.

Since there's many GPU .so files in the same folder typically, this would mean we only have to scan once per directory. Storing all of the symlinks in memory for these paths will not use that much memory

Reply:
Done!